git_machete/client.py

```diff
@@ -2362,3 +2386,23 @@ def __sync_before_creating_pr(self, *, opt_onto: Optional[LocalBranchShortName],
             if ans in ('y', 'yes'):
                 return
             raise MacheteException('Pull request creation interrupted.')
+
+    def delete_untracked(self,
+                         opt_yes: bool,
+                         opt_merge: bool,
+                         opt_no_interactive_rebase: bool,
+                         opt_no_edit_merge: bool) -> None:
+        untracked_branches: List[LocalBranchShortName] = []
+        for managed_branch in self.managed_branches:
+            status, _ = self.__git.get_strict_remote_sync_status(managed_branch)
+            if status == SyncToRemoteStatuses.UNTRACKED:
+                untracked_branches.append(managed_branch)
+
+        if untracked_branches:
+            self._slide_out_branches_from_git_machete(branches_to_slide_out=untracked_branches,
+                                                      opt_yes=opt_yes,
+                                                      opt_merge=opt_merge,
+                                                      opt_no_interactive_rebase=opt_no_interactive_rebase,
+                                                      opt_no_edit_merge=opt_no_edit_merge,
+                                                      delete_branches=True)
+            self.managed_branches = excluding(self.managed_branches, untracked_branches)
```

Comment:
Huh this should as well be sorted out by `_slide_out_branches_from_git_machete`... doesn't that method update `self.managed_branches` adequately?

Reply:
No, it doesn't

Reply:
I'd make sure it does then! so that `_slide_out_branches_from_git_machete` gets everything sorted out on its own... in fact I'd even extract this refactor as a separate PR, to be used as a base PR for this one